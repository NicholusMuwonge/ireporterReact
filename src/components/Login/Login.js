/* eslint-disable react/prop-types */
import React, { Fragment } from 'react';
import NavBar from '../navbar/Navbar';

const Login = ({
  username, submitHandler, password, handleChange, buttonLoading,
}) => (
  <Fragment>
    <NavBar />
    <div className="signup-container">
      <div className="container">
        <div className="row">
          <div className="col-lg-10 col-xl-9 mx-auto">
            <div className="card card-signin flex-row my-5">
              <div className="card-img-left d-none d-md-flex" />
              <div className="card-body">
                <h5 className="card-title text-center">Login</h5>
                <form className="form-signin" onSubmit={submitHandler}>
                  <div className="form-label-group">
                    <span htmlFor="inputUserame">
                            Username
                      <input type="text" id="inputUserame" name="userName" className="form-control" onChange={handleChange} placeholder="Username" value={username} required />
                    </span>
                  </div>
                  <hr />
                  <div className="form-label-group">
                    <span>
                          Password
                      <input type="password" id="inputConfirmPassword" name="password" className="form-control" onChange={handleChange} placeholder="Password" value={password} required />

                    </span>
                  </div>
                  <button className="btn btn-lg btn-primary btn-block text-uppercase" type="submit" disabled={buttonLoading}>
                    {buttonLoading && (<i className="spinner-grow spinner-grow-sm" />)}
                    {buttonLoading && <span>Just A sec Please!...</span>}
                    {!buttonLoading && <span>Login</span>}
                  </button>
                  <a className="d-block text-center mt-2 small" href="/signup">Sign Up</a>
                  <hr className="my-4" />
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Fragment>
);


export default Login;
