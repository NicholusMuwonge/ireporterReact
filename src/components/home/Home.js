import React, { Component, Fragment } from 'react';

class Home extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    return (
      <Fragment>
        <header>
          <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
            <ol className="carousel-indicators">
              <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active" />
              <li data-target="#carouselExampleIndicators" data-slide-to="1" />
              <li data-target="#carouselExampleIndicators" data-slide-to="2" />
            </ol>
            <div className="carousel-inner" role="listbox">
              {/* <!-- Slide One - Set the background image for this slide in the line below --> */}
              <div
                className="carousel-item active"
                style={{ backgroundImage: "url('https://source.unsplash.com/KSdfplrJreg/1920x720')" }}
              >
                <div className="carousel-caption d-none d-md-block">
                  <h1 className="display-4">Community Incidents</h1>
                  <p className="">Get to raise the issues that impact your community</p>
                </div>
              </div>
              {/* <!-- Slide Two - Set the background image for this slide in the line below --> */}
              <div
                className="carousel-item"
                style={{ backgroundImage: "url('https://source.unsplash.com/udXD2NrbXS8/1920x720')" }}
              >
                <div className="carousel-caption d-none d-md-block">
                  <h2 className="display-4">Raise Corruption Related Crimes</h2>
                  <p className="lead">These are sensitive issues that hold your community aback, so raise them right away</p>
                </div>
              </div>
              <div
                className="carousel-item"
                style={{ backgroundImage: "url('https://source.unsplash.com/lbLgFFlADrY/1920*720')" }}
              >
                <div className="carousel-caption d-none d-md-block">
                  <h2 className="display-4">Reconcile All the Community Differences</h2>
                  <p className="lead">We get further together and in Unisson compared to being of different sides.</p>
                </div>
              </div>
            </div>
            <a className="carousel-control-prev" href="#carouselExampleIndicators" role="button" data-slide="prev">
              <span className="carousel-control-prev-icon" aria-hidden="true" />
              <span className="sr-only">Previous</span>
            </a>
            <a className="carousel-control-next" href="#carouselExampleIndicators" role="button" data-slide="next">
              <span className="carousel-control-next-icon" aria-hidden="true" />
              <span className="sr-only">Next</span>
            </a>
          </div>
        </header>
      </Fragment>
    );
  }
}


export default Home;
