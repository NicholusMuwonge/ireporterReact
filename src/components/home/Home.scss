.carousel-item {
    height: 100vh;
    min-height: 350px;
    background: no-repeat center center scroll;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
  }

  
#navbarResponsive{
    padding: 30px;
}

#cream{
    background-color: antiquewhite;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.login{
    color: black;
    background-color: none;
    font-weight: bolder;
    border: 1px solid teal;
}

.login:hover{
    background-color: teal;
    color: white;
}

.signup{
    background-color:none;
    border: 1px solid purple;
    color:black;
    margin-left: left;
    font-weight: bolder;
}

.signup:hover{
    background-color: purple;
    color: white;
}

.title{
    font-size: xx-large;
    color: rgb(17, 11, 11);
    text-decoration: underline teal;

}