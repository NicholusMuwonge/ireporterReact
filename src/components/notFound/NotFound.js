import React from 'react';
import NavBar from '../navbar/Navbar';

const notFound = () => (
  <>
    <NavBar />
    <h1 className="notFound">Page Not found</h1>
  </>
);

export default notFound;
