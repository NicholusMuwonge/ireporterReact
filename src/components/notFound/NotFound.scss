.notFound {
  margin-left: 20%;
  margin-top: 17%;
  color: teal;
  font-size: 8em;
}
