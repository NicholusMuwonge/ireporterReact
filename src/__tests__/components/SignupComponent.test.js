import React from 'react';
import { shallow } from 'enzyme';
import Login from '../../components/Login/Login';
import SignUp from '../../components/Signup/Signup';
import Home from '../../components/home/Home';
import NavBar from '../../components/navbar/Navbar';
import notFound from '../../components/notFound/NotFound';


describe('<Login />', () => {
  let wrapper;
  const props = {
    state: {
      data: {
        username: 'first',
      },
    },
    data: { },
  };
  it('it should render component without fail', () => {
    wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render component without fail', () => {
    wrapper = shallow(<SignUp {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render component without fail', () => {
    wrapper = shallow(<Home {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render component without fail', () => {
    wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should render component without fail', () => {
    wrapper = shallow(<notFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
