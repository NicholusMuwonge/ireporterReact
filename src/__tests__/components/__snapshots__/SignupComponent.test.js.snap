// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> it should render component without fail 1`] = `ShallowWrapper {}`;

exports[`<Login /> it should render component without fail 2`] = `ShallowWrapper {}`;

exports[`<Login /> it should render component without fail 3`] = `ShallowWrapper {}`;

exports[`<Login /> it should render component without fail 4`] = `ShallowWrapper {}`;

exports[`<Login /> it should render component without fail 5`] = `ShallowWrapper {}`;
